flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/aggregate/window/combines/CombineRecordsFunction.java

```diff
@@ -59,8 +57,11 @@
     /** Whether to copy key and input record, because key and record are reused. */
     private final boolean requiresCopy;
 
+    /** Serializer to copy key if required. */
+    private AbstractRowDataSerializer<RowData> keySerializer;
```

Comment:
Can be final. And I think `TypeSerializer<RowData>` is enough.  The same to others.

Reply:
👍 Now types are:
```
    private AbstractRowDataSerializer<RowData> inputSerializer; // Need toBinaryRow
    private PagedTypeSerializer<RowData> keySerializer; // Need PagedTypeSerializer in BytesHashMap
    private TypeSerializer<RowData> accSerializer;
```